zcash_client_sqlite/src/lib.rs

```diff
@@ -2020,6 +2074,111 @@ mod tests {
+            .with_account_from_sapling_activation(BlockHash([0; 32]))
+            .with_account_having_index(zip32::AccountId::ZERO)
+            .build();
+        assert_matches!(
+            st.test_account().unwrap().account().source(),
+            AccountSource::Derived { seed_fingerprint: _, account_index } if account_index == zip32::AccountId::ZERO);
+    }
+
+    #[test]
+    pub(crate) fn import_account_hd_1_then_2() {
+        let mut st = TestBuilder::new().build();
+
+        let birthday = AccountBirthday::from_parts(
+            ChainState::empty(st.wallet().params.sapling.unwrap() - 1, BlockHash([0; 32])),
+            None,
+        );
+
+        let seed = Secret::new(vec![0u8; 32]);
+        let zip32_index = zip32::AccountId::ZERO.next().unwrap();
+
+        let first = st
+            .wallet_mut()
+            .import_account_hd(&seed, zip32_index, &birthday)
+            .unwrap();
+        assert_matches!(
+            first.0.source(),
+            AccountSource::Derived { seed_fingerprint: _, account_index } if account_index == zip32_index);
+
+        let second = st
+            .wallet_mut()
+            .import_account_hd(&seed, zip32_index.next().unwrap(), &birthday)
+            .unwrap();
+        assert_matches!(
+            second.0.source(),
+            AccountSource::Derived { seed_fingerprint: _, account_index } if account_index == zip32_index.next().unwrap());
+    }
+
+    #[test]
+    pub(crate) fn import_account_hd_1_twice() {
+        let mut st = TestBuilder::new().build();
+
+        let birthday = AccountBirthday::from_parts(
+            ChainState::empty(st.wallet().params.sapling.unwrap() - 1, BlockHash([0; 32])),
+            None,
+        );
+
+        let seed = Secret::new(vec![0u8; 32]);
+        let zip32_index = zip32::AccountId::ZERO.next().unwrap();
+
+        let first = st
+            .wallet_mut()
+            .import_account_hd(&seed, zip32_index, &birthday)
+            .unwrap();
+
+        assert_matches!(
+            st.wallet_mut().import_account_hd(&seed, zip32_index, &birthday),
+            Err(SqliteClientError::AccountCollision(id)) if id == first.0.id());
+    }
+
+    #[test]
+    pub(crate) fn import_account_ufvk() {
+        let mut st = TestBuilder::new().build();
+
+        let birthday = AccountBirthday::from_parts(
+            ChainState::empty(st.wallet().params.sapling.unwrap() - 1, BlockHash([0; 32])),
+            None,
+        );
+
+        let seed = vec![0u8; 32];
+        let usk = UnifiedSpendingKey::from_seed(&st.wallet().params, &seed, zip32::AccountId::ZERO)
+            .unwrap();
+        let ufvk = usk.to_unified_full_viewing_key();
+
+        let account = st
+            .wallet_mut()
+            .import_account_ufvk(&ufvk, &birthday, true)
+            .unwrap();
+        assert_eq!(
+            ufvk.encode(&st.wallet().params),
+            account.ufvk().unwrap().encode(&st.wallet().params)
+        );
```

Comment:
Check that the resulting `account` is imported, not derived (like the earlier tests check the reverse).